src/main/java/com/xebialabs/overthere/ssh/SshSftpFile.java

```diff
@@ -137,6 +137,24 @@ protected FileAttributes stat() {
     }
 
     @Override
+    protected void copyFrom(OverthereFile source) {
+        SFTPFileTransfer fileTransfer = connection.getSharedSftpClient().getFileTransfer();
+        try {
+            // If the source is a directory and the name and the target name do not match, SFTP copies the
+            // directory into the destination, whereas we want to copy the contents into the destination.
+            if (source.isDirectory() && this.exists() && !source.getName().equals(this.getName())) {
```

Comment:
Do we need to take into account issues related to the case of the name here? E.g. an SSH implementation running on Windows?